src/tests/ftest/launch.py

```diff
@@ -2965,131 +2982,136 @@ def __init__(self, log):
+            for core_name in os.listdir(core_dir):
+                if not fnmatch.fnmatch(core_name, 'core.*[0-9]'):
+                    continue
+                core_files[core_dir].append(core_dir)
 
-        # Copy any core files that exist on the test hosts and remove them from the
-        # test host if the copy is successful.  Attempt all of the commands and
-        # report status at the end of the loop.  Include a listing of the file
-        # related to any failed command.
-        commands = [
-            "set -eu",
-            "rc=0",
-            "copied=()",
-            "df -h /var/tmp",
-            "for file in /var/tmp/core.*",
-            "do if [ -e $file ]",
-            "then ls -al $file",
-            "if [ ! -s $file ]",
-            "then ((rc++))",
-            "else if sudo chmod 644 $file && "
-            f"scp $file {get_local_host()}:{daos_cores_dir}/${{file##*/}}-$(hostname -s)",
-            "then copied+=($file)",
-            "if ! sudo rm -fr $file",
-            "then ((rc++))",
-            "fi",
-            "else ((rc++))",
-            "fi",
-            "fi",
-            "fi",
-            "done",
-            "echo Copied ${copied[@]:-no files}",
-            "exit $rc",
-        ]
-        result = run_remote(self.log, test_hosts, "; ".join(commands), timeout=1800)
-        if not result.passed:
-            # we might have still gotten some core files, so don't return here
-            # but save a False return status for later
-            status = False
+        # Install the debug information needed for stacktrace generation
+        if core_files:
+            try:
+                self.install_debuginfos()
+            except LaunchException as error:
+                self.log.error(error)
+                self.log.debug("Stacktrace", exc_info=True)
+                errors += 1
+        else:
+            self.log.debug(
+                "No core.*[0-9] files found in %s", os.path.join(directory, "stacktraces*"))
 
-        cores = os.listdir(daos_cores_dir)
-        if not cores:
-            return True
+        # Create a stacktrace from each core file
+        if not self.is_el7() and errors == 0:
+            for core_dir, core_name_list in core_files.items():
+                for core_name in core_name_list:
+                    try:
+                        self._create_stacktrace(core_dir, core_name)
+                    except Exception as error:      # pylint: disable=broad-except
+                        self.log.error(error)
+                        self.log.debug("Stacktrace", exc_info=True)
+                        errors += 1
+
+        # Remove the stacktrace files
+        for core_dir, core_name_list in core_files.items():
+            for core_name in core_name_list:
+                core_full = os.path.join(core_dir, core_name)
+                self.log.debug("Removing %s", core_full)
+                os.remove(core_full)
```

Comment:
Remove core file, not stacktrace, right?
Also, `core_full` -> `core_file`?

Reply:
Fixed